ceph/rados/mgr_workflows.py

```diff
@@ -0,0 +1,294 @@
+
+        additional_mgr_count = new_mgr_count - mgr_count
+        if mgr_count == new_mgr_count:
+            log.info(" The request count is same as the current mgr count")
+            if set(nodes_list) == set(mgr_node_list):
+                log.info("The new mgr nodes list is same as old list")
+            return True
+
+        new_mgr = []
+        if new_mgr_count > mgr_count:
+            log.info(
+                f"Performing MGR scale up.The mgr is scale-up from {mgr_count} to {new_mgr_count} "
+            )
+            scale_up = True
+            # Node list to add the label
+            new_mgr = mgr_node_list[-additional_mgr_count:]
+        else:
+            log.info(
+                f"Performing MGR scale down.The mgr is scale-down from {mgr_count} to { new_mgr_count} "
+            )
+            # Node list to remove the label
+            new_mgr_daemon = [
+                item
+                for item in mgr_daemons
+                if not item.startswith(tuple(mgr_node_list))
+            ]
+            for item in new_mgr_daemon:
+                new_mgr.append(item.split(".")[0])
+            scale_up = False
+
+        mgr_nodes = "  ".join(mgr_node_list)
+        cmd = f'ceph orch apply mgr "{new_mgr_count}  {mgr_nodes}"'
+        self.client.exec_command(sudo=True, cmd=cmd)
+
+        if scale_up:
+            for node in new_mgr:
+                cmd = f"ceph orch host label add {node} mgr"
+                self.client.exec_command(sudo=True, cmd=cmd)
+                log.debug(f"Added mgr label for the host : {node}")
+                time.sleep(10)
+        else:
+            for node in new_mgr:
+                cmd = f"ceph orch host label rm {node} mgr"
+                self.client.exec_command(sudo=True, cmd=cmd)
+                log.debug(f"Removed mgr label for the host : {node}")
+                time.sleep(10)
+
+        endtime = datetime.datetime.now() + datetime.timedelta(seconds=180)
+        time_flag = False
+        while datetime.datetime.now() < endtime:
+            mgr_count = 0
+            for node in mgr_node_list:
+                if self.check_mgr_exists_on_host(host=node):
+                    log.info(f"Mgr {node} is up and running in the cluster.")
+                    mgr_count = mgr_count + 1
+                    if new_mgr_count == mgr_count:
+                        log.info(" All mgr nodes to  up and running")
+                        time_flag = True
+                        break
+                else:
+                    log.info(f"Mgr {node} is not running in the cluster.")
+                    time.sleep(5)
+            if time_flag:
+                log.info(f"All mgrs are running and not checking up to {endtime}")
+                break
+        if mgr_count != new_mgr_count:
+            log.info(f" The {mgr_node_list} are not up and running")
+            return False
+        return True
+
+    def check_mgr_exists_on_host(self, host) -> bool:
```

Comment:
This can be generalized for any daemons. Collect the daemon type and host name, and the method would return if the given daemon type is present or not on the cluster.

Should be fine for now, but please have a backlog task for modifying the method implementation.

Reply:
Created the following backlog task to update the code.
https://issues.redhat.com/browse/RHCEPHQE-14837

Reply:
Created a general method in the core_workflows.py